tests/unit/components/generic/CompareImageSlider.spec.js

```diff
@@ -0,0 +1,86 @@
+import { createLocalVue, shallowMount } from '@vue/test-utils';
+import BootstrapVue from 'bootstrap-vue';
+import sinon from 'sinon';
+import CompareImageSlider from '../../../../components/generic/CompareImageSlider.vue';
+
+const localVue = createLocalVue();
+localVue.use(BootstrapVue);
+
+const factory = () => shallowMount(CompareImageSlider, {
```

Comment:
to test the setImageWidth call do this:
```
const factory = (setImageWidthSpy) => shallowMount(CompareImageSlider, {
  localVue,
  propsData: {
    imageLeft: 'img/landscape.jpg',
    imageRight: 'img/portrait.jpg'
  },
  methods: { setImageWidth: setImageWidthSpy }
});

describe('components/generic/CompareImageSlider', () => {

  it('calls `setImageWidth` method when browser has been resized', () => {
    const setImageWidth = sinon.spy();
    factory(setImageWidth);

    global.window.dispatchEvent(new Event('resize'));

    setImageWidth.should.have.callCount(2);
  });

  ...
```
This won't execute the actual function any more, but sets up the spy before mounting, so we can refer to it later. If this interferes with the other tests, then we could have a specific factory definition just for that one example.

Reply:
BOOM!!!! Thank you Lutz. Works a treat :)